tests/test_generics.py

```diff
@@ -437,16 +437,15 @@ class Result(BaseModel, Generic[data_type, error_type]):
         positive_number: int
 
         @validator('error')
-        def validate_error(cls, v: Optional[error_type], **kwargs) -> Optional[error_type]:
```

Comment:
@dmontagu can you confirm this change is right since you just wrote the test?

Reply:
as far as I can tell this looks good to me, with one exception (that I don't think is your fault): below, it copies this twice:
```python

    with pytest.raises(ValidationError) as exc_info:
        Result[Data, Error](data=[Data(number=1, text='a')], error=Error(message='error'), positive_number=1)
    assert exc_info.value.errors() == [
        {
            'type': 'value_error',
            'loc': ('error',),
            'msg': 'Value error, Must provide data or error',
            'input': Error(message='error'),
            'ctx': {'error': 'Must provide data or error'},
        }
    ]
```

I think the second time, it should be more like:
```python

    with pytest.raises(ValidationError) as exc_info:
        Result[Data, Error](positive_number=1)
    assert exc_info.value.errors() == [
        {
            'type': 'value_error',
            'loc': ('error',),
            'msg': 'Value error, Must not provide both data and error',
            'input': Error(message='error'),
            'ctx': {'error': 'Must not provide both data and error'},
        }
    ]
```

Reply:
removed the duplication